### PR TITLE
Update timetable/teacher models

### PR DESCRIPTION
**Pull request by Ts1n**
- Update models according to I-On Core specifications
&nbsp;&nbsp;&nbsp;&nbsp; `Language` and `Category` properties added
- Update builder and test to factor this changes
- Fix regex that didn't match some programmes pdf's

### CODE REVIEW

File: src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt [hunk 1]
```diff
@@ -200,15 +207,22 @@ class IselTimetableTeachersBuilder() : ITimetableTeachersBuilder<RawData> {
...
     }
 
+    private fun getDescription(classType: String) = when (classType) {
+        "T" -> "Aulas Teóricas de "
+        "P" -> "Aulas Práticas de "
+        "L" -> "Aulas Laboratório de "
+        "T/P" -> "Aulas Teórico-práticas de "
+        else -> ""
+    }
+
```
Comment (Ts1n) [inline src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt:226]: @pmhsfelix Do you agree with this values?

File: src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.ionproject.integration.model.internal.timetable
+
+enum class EventCategory {
+    CLASS {
+        override fun toString() = "AULA"
```
Comment (Ts1n) [inline src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt]: Still waiting for I-On Core to confirm this

File: src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt (hunk 1 above)
Comment (pmhsfelix) [reply src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt:226]: Where is this used? I think, so for the moment being. We probably need to revisit this but that will be done "next year".

File: src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt (hunk 2 above)
Comment (pmhsfelix) [reply src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt]: @joao-pereira5 FYI

**Review by pmhsfelix — APPROVED** [review]

**Review by sscosta — APPROVED** [review]

File: src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt (hunk 2 above)
Comment (jmchamica) [reply src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt]: The identifiers for event categories are defined [here](https://github.com/i-on-project/core/blob/master/docs/api/write/insertClassSectionEvents.md#constants).

File: docs/batch jobs/ISEL timetable extraction batch job/isel-timetable-batch-specification.md [hunk 3]
```diff
@@ -74,7 +76,7 @@ data class Label(
 data class Event (
 	@JsonProperty("title") val title : String,
     @JsonProperty("description") val description : String,
-    @JsonProperty("type") val type : String,
+    @JsonProperty("category") val category : String,
```
Comment (jmchamica) [inline docs/batch jobs/ISEL timetable extraction batch job/isel-timetable-batch-specification.md]: According to the [documentation](https://github.com/i-on-project/core/blob/master/docs/api/write/insertClassSectionEvents.md), the category would be a positive integer (unique ID). The available values are listed [here](https://github.com/i-on-project/core/blob/master/docs/api/write/insertClassSectionEvents.md#constants).

File: src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt [hunk 4]
```diff
@@ -93,11 +94,13 @@ class IselTimetableTeachersBuilder() : ITimetableTeachersBuilder<RawData> {
         timetable.programme = Programme(name = programme)
         timetable.calendarTerm = calendarTerm
         timetable.calendarSection = classSection
+        timetable.language = "pt-PT"
 
         courseTeacher.school = School(name = school)
         courseTeacher.programme = Programme(name = programme)
         courseTeacher.calendarTerm = calendarTerm
         courseTeacher.calendarSection = classSection
+        courseTeacher.language = "pt-PT"
```
Comment (jmchamica) [inline src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt]: The values which identify languages will be the same for all write API operations and [they are listed here](https://github.com/i-on-project/core/blob/master/docs/api/write/insertClassSectionEvents.md#constants).

File: src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt (hunk 2 above)
Comment (jmchamica) [reply src/main/kotlin/org/ionproject/integration/model/internal/timetable/EventCategory.kt]: To further clarify, in this case, the lecture should be marked with `2`.

**Pull request merged**